models/issue.go

```diff
@@ -415,6 +416,22 @@ func (issue *Issue) HasLabel(labelID int64) bool {
 	return issue.hasLabel(db.GetEngine(db.DefaultContext), labelID)
 }
 
+// ReplyReference returns tokenized address to use for email reply headers
+func (issue *Issue) ReplyReference(key string) string {
+	var path string
+	if issue.IsPull {
+		path = "pulls"
+	} else {
+		path = "issues"
+	}
+
+	if len(key) > 0 {
+		return fmt.Sprintf("%s/%s/%d?%s@%s", issue.Repo.FullName(), path, issue.Index, key, setting.Domain)
```

Comment:
https://datatracker.ietf.org/doc/html/rfc2392 

```
     message-id    = url-addr-spec
     url-addr-spec = addr-spec  ; URL encoding of RFC 822 addr-spec
```

The `key` might need to be encoded.

Reply:
But this format is too strict to produce a meaningfull messgae, I wonder.
```  
     addr-spec   =  local-part "@" domain        ; global address
     
     local-part  =  word *("." word)             ; uninterpreted
                                                 ; case-preserved
```

example of this format:
```
12343@domain
22233.word.123@domain
```

and github also not use this format:
![image](https://user-images.githubusercontent.com/25342410/141666108-b8b6870a-50dc-47d0-ae4b-df9f3dff3a64.png)

Reply:
An word is either an atom or a quoted string.

Positively speaking, this means that the valid constituents of an atom are the following:

```
!"#$%&'*+-/
0123456789
=?@
ABCDEFGHIJKLMNOPQRSTUVWXYZ
^_`
abcdefghijklmnopqrstuvwxyz
{|}~
```

Not that strict